src/skin/legacyskinparser.cpp

```diff
@@ -361,7 +362,12 @@ QWidget* LegacySkinParser::parseSkin(const QString& skinPath, QWidget* pParent)
         }
 
         if (created) {
-            created_attributes.append(pControl);
+            // If we created this control, add it to the set of skin-created
+            // controls, so that it can be deleted when the MixxxMainWindow is
+            // destroyed.
+            VERIFY_OR_DEBUG_ASSERT(m_pSkinCreatedControls) {
```

Comment:
No, this doesn't work as intended.

And m_pSkinCreatedControls is never supposed to be NULL. This could be asserted in the constructor. If it is NULL let it crash. We only need to pass a pointer by convention, although it actually is mutable reference that never becomes NULL.

Reply:
:blush: I always get confused by VERIFY_OR_DEBUG_ASSERT because it looks like an if condition but the logic is the opposite of it.